terraform/modules/aws/route53/main.tf

```diff
@@ -0,0 +1,47 @@
+resource "aws_route53_zone" "bpzb" {
+  name = var.zone_name
+}
+
+resource "aws_route53_record" "api_bpzb_a" {
+  name    = var.a_record_name
+  type    = var.a_record_type
+  zone_id = aws_route53_zone.bpzb.zone_id
+  alias {
+    name                   = var.route53_alb_dns_name
+    zone_id                = var.route53_alb_zone_id
+    evaluate_target_health = var.a_target_health
+  }
+  depends_on = [aws_route53_zone.bpzb]
+}
+
+resource "aws_acm_certificate" "api_bpzb" {
+  domain_name       = var.cert_domain_name
+  validation_method = var.cert_validation_method
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_route53_record" "api_bpzb_cname" {
+  for_each = {
+    for dvo in aws_acm_certificate.api_bpzb.domain_validation_options : dvo.domain_name => {
```

Comment:
seems like we do not need for each here  because aws_acm_certificate is a single resource

Reply:
single, but dvo is a list of objects
       + domain_validation_options = [
          + {
              + domain_name           = "api.bpzb.smodata.net"
              + resource_record_name  = (known after apply)
              + resource_record_type  = (known after apply)
              + resource_record_value = (known after apply)
            },
        ]

Reply:
ok. Can we use var.records for this record too?

Reply:
yes

Reply:
Not bad man!